.github/workflows/plugin-upgrade-test.yml

```diff
@@ -15,40 +15,65 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  validate-changelog-diff:
+    name: Validate Changelog Diff
+    runs-on: ubuntu-22.04
+    outputs:
+      changelog_diff: ${{ steps.changelog_diffs.outputs.changelog_diff }}
+    steps:
+      - name: Checkout Source Code
+        uses: actions/checkout@v4
+
+      - name: Check if this is only a changelog PR
+        id: changelog_diffs
+        uses: ./.github/workflows/check-changelog-pr
+
   build-plugin:
-    uses: ./.github/workflows/build.yml
+    name: Playwright plugin upgrade Build Plugin
+    needs: [ validate-changelog-diff ]
+    if: ${{ github.event.pull_request.title == null || needs.validate-changelog-diff.outputs.changelog_diff }}
+    uses: ./.github/workflows/plugin-build.yml
+
   run-upgrade-test:
     name: Playwright plugin upgrade test
     runs-on: ubuntu-latest
-    needs: [build-plugin]
-    if: ${{ github.event.pull_request.title == null || needs.build-plugin.outputs.changelog_diff }}
+    needs: [build-plugin, validate-changelog-diff]
+    if: ${{ github.event.pull_request.title == null || needs.validate-changelog-diff.outputs.changelog_diff }}
```

Comment:
I think that all the `changelog_diff` conditions should be the same.
Please go over all of them and unify them id necessary.